lib/vim-mode.coffee

```diff
@@ -33,8 +33,18 @@ module.exports =
   deactivate: ->
     @disposables.dispose()
 
+  getGlobalState: ->
+    @globalVimState
+
+  getState: (editor) ->
+    @vimStates.get(editor)
+
   consumeStatusBar: (statusBar) ->
     @statusBarManager.initialize(statusBar)
     @statusBarManager.attach()
     @disposables.add new Disposable =>
       @statusBarManager.detach()
+
+  provideVimMode: ->
+    getGlobalState: @getGlobalState.bind(@)
+    getState: @getState.bind(@)
```

Comment:
I think I'd rather have this be called `getEditorState` to make it clear that you need to pass a `TextEditor`. You cool w/ that @lloeki?